File: tournaments/allLongestStrings/allLongestStrings.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub allLongestStrings {
    my ($inputArray) = @_;

    my @result = ();

    foreach (@$inputArray) {
        if (!scalar(@result) || length($result[0]) == length($_)) {
            push @result, $_;
        } elsif (length($result[0]) < length($_)) {
            @result = ($_);
        }
    }
    return \@result;
}

File: tournaments/allLongestStrings/allLongestStrings.js
function allLongestStrings(inputArray) {
    const result = [];

    for (const item of inputArray) {
        if (!result.length || result[0].length === item.length) {
            result.push(item);
        } else if (result[0].length < item.length) {
            result.splice(0, result.length);
            result.push(item);
        }
    }

    return result;
}
